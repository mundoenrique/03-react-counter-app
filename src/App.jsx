import { Fragment } from 'react'
import PropTypes from 'prop-types'
import './App.css'

const name = 'Enrique';
const arr = [1, 2, 3, 4, 5, 6];
const obj = {
	name: 'yajaira',
	edad: 40
}

function addNumbers(a, b) {
	return a + b;
}

export default function App({ title, subTitle }) {

	return (
		<Fragment>
			<h1>{title} {name}</h1>
			<h2>{subTitle}</h2>
			<p>Arreglo {arr}</p>
			<p>Sumar números {addNumbers(4, 5)}</p>
			<pre>Objeto = {JSON.stringify(obj, null, 2)}</pre>
		</Fragment>
	)
}

App.propTypes = {
	title: PropTypes.string.isRequired,
	subTitle: PropTypes.string,
}

App.defaultProps = {
	title: 'Sin título',
	subTitle: 'Aprendiendo React'
}
