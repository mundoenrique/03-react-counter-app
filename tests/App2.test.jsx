import { render, screen } from '@testing-library/react';
import App from '../src/App';

describe('Pruebas en <App />', () => {
	const title = 'Bienvenido';
	const subTitle = 'Aprendiendo React con vite';

	test('Debe hacer match con el snapshot', () => {
		const { container } = render(<App title={title} />);

		expect(container).toMatchSnapshot();
	});

	test('Debe mostrar el mensaje "Bienvenido Enrique"', () => {
		render(<App title={title} />);

		expect(screen.getByText(`${title} Enrique`)).toBeTruthy();
	});

	test('Debe mostrar el mensaje "Bienvenido Enrique" en <h1>', () => {
		render(<App title={title} />);

		expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(
			`${title} Enrique`
		);
	});

	test('Debe mostrar el subtitulo', () => {
		render(<App title={title} subTitle={subTitle} />);

		expect(screen.getByText(subTitle)).toBeTruthy();
		expect(screen.getAllByText(subTitle).length).toBe(1);
	});
});
