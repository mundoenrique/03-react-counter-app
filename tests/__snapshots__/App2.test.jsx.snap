// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <App /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Bienvenido
     
    Enrique
  </h1>
  <h2>
    Aprendiendo React
  </h2>
  <p>
    Arreglo 
    1
    2
    3
    4
    5
    6
  </p>
  <p>
    Sumar n√∫meros 4 + 5 
    9
  </p>
  <pre>
    Objeto = 
    {
  "name": "yajaira",
  "edad": 40
}
  </pre>
</div>
`;
